cmd/system-probe/modules/network_tracer.go

```diff
@@ -78,13 +80,79 @@ type networkTracer struct {
 	tracer       *tracer.Tracer
 	done         chan struct{}
 	restartTimer *time.Timer
+
+	// Intended to ensure (in compile time) we're implementing the SystemProbeServer interface
+	connectionserver.UnsafeSystemProbeServer
+	maxConnsPerMessage int
 }
 
 func (nt *networkTracer) GetStats() map[string]interface{} {
 	stats, _ := nt.tracer.GetStats()
 	return stats
 }
 
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// GetConnections function that establishes a streaming RPC connection to retrieve and continuously stream information
+// about the current connections in the system.
+func (nt *networkTracer) GetConnections(req *connectionserver.GetConnectionsRequest, s2 connectionserver.SystemProbe_GetConnectionsServer) error {
+	start := time.Now()
+	runCounter := atomic.NewUint64(0)
+	id := req.GetClientID()
+	cs, err := nt.tracer.GetActiveConnections(id)
+	if err != nil {
+		return err
+	}
+	defer network.Reclaim(cs)
+
+	marshaler := encoding.GetMarshaler(encoding.ContentTypeProtobuf)
+	connectionsModeler := encoding.InitConnectionsModeler(cs)
+	if nt.restartTimer != nil {
+		nt.restartTimer.Reset(inactivityRestartDuration)
+	}
+	count := runCounter.Inc()
+	logRequests(id, count, len(cs.Conns), start)
+	connections := &connectionserver.Connection{}
+
+	// As long as there are connections, we divide them into batches and subsequently send all the batches
+	// via a gRPC stream to the process agent. The size of each batch is determined by the value of maxConnsPerMessage.
+	for len(cs.Conns) > 0 {
+		finalBatchSize := min(nt.maxConnsPerMessage, len(cs.Conns))
+		rest := cs.Conns[finalBatchSize:]
+		cs.Conns = cs.Conns[:finalBatchSize]
+
+		payload := connectionsModeler.ModelConnections(cs)
+		// get the conns for a batch by the marshaler
+		conns, err := marshaler.Marshal(payload)
+		encoding.Cleanup(payload)
+
+		if err != nil {
+			return err
```

Comment:
suggestion: add some small amount of context for the error here, to differentiate between all the possible errors in this function. Example `return fmt.Errorf("marshal: %s", err)`